### PR TITLE
Backend for Registration

### PR DESCRIPTION
**Pull request by PeterDMadrid**
**OVERVIEW**

- Registration Validation
- Email Verification
- Test pages inside views

**SNIPPETS**
Test form:
Form

When submitted:
testwait

Verify Email:
verify

Proceed to another test page after verifying email:
testpage

### CODE REVIEW

**Review by adPalafox — COMMENTED** [review]
Screenshot 2024-04-11 at 17 59 21
Pls add new line when you see the red "-" sign on file changed. Thank you

File: src/app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                    $request->zip_code,
+                ])),
+            ]);
+
+            auth()->login($user);
+            event(new Registered($user));
+
+            return redirect('/verify-waiting')->with('success', 'Registration is successful! Please check your email for verification.');
+        } catch (\Exception $e) {
+            dd($e->getMessage());
```
Comment (sunkenquest) [inline src/app/Http/Controllers/AuthController.php]: handle the error appropriately, maybe displaying the error on the frontend not just using `dd`

File: src/app/Models/Profile.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        'first_name',
+        'last_name',
+        'middle_name',
+        'birth_date',
+        'address'
+    ];
+    /**
+     * Get the attributes that should be cast.
+     *
+     * @return int
```
Comment (sunkenquest) [inline src/app/Models/Profile.php:23]: This is good.

File: src/app/Http/Controllers/AuthController.php (hunk 1 above)
Comment (PeterDMadrid) [reply src/app/Http/Controllers/AuthController.php]: Replaced dd by displaying the error on frontend. Also added error bag in views.

**Review by sunkenquest — APPROVED** [review]
GEFN 🙇🏼

**Pull request merged**